src/analysis/processing/qgsalgorithmrasterize.cpp

```diff
@@ -150,6 +152,76 @@ QVariantMap QgsRasterizeAlgorithm::processAlgorithm( const QVariantMap &paramete
   int height { yTileCount * tileSize };
   int nBands { transparent ? 4 : 3 };
 
+  int totalTiles = 0;
+  for ( auto &layer : std::as_const( mMapLayers ) )
+  {
+    if ( QgsRasterLayer *rasterLayer = dynamic_cast<QgsRasterLayer *>( ( layer.get() ) ) )
+    {
+      if ( rasterLayer->dataProvider() )
+      {
+        QgsProviderMetadata *metadata = QgsProviderRegistry::instance()->providerMetadata( rasterLayer->dataProvider()->name() );
```

Comment:
What do you feel about adding a providerMetadata method to QgsMapLayer? It'd simplify code like this a lot

Reply:
@nyalldawson , attached to map layer or attached to its data provider?

Reply:
I'd say map layer